### PR TITLE
New script for customers to validate partitioned graph objects

### PR DESCRIPTION
**Pull request by kylasa**
## Description

Following changes are implemented with this PR. 

1. A new script (tools/verify_partitions.py) is added so that customers can run and validate the partitioned graph objects with the original graph.
2. Moved all the common code to tools/verification_utils.py and functions are invoked from tools/verify_partitions.py and tests/tools/test_dist_part.py
3. Added new method to check the degrees of nodes from the partitioned graph objects and compared them with the original graph.
4. Existing tests, using _test_basic_pipeline, is modified to either validate the graph objects using a local code or use the new script (tools/verify_partitions.py).

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [X] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [X] All changes have test coverage
- [X] Code is well-documented
- [X] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/tools/test_dist_part.py [hunk 1]
```diff
@@ -222,6 +166,13 @@ def test_chunk_graph_vector_rows(num_chunks, vector_rows):
     )
 
 
+"""
```
Comment (Rhett-Ying) [inline tests/tools/test_dist_part.py:169]: remove deprecated code

File: tests/tools/test_dist_part.py [hunk 2]
```diff
@@ -367,6 +332,13 @@ def test_pipeline_formats(graph_formats):
     _test_pipeline(4, 4, 4, graph_formats)
 
 
+"""
```
Comment (Rhett-Ying) [inline tests/tools/test_dist_part.py:335]: remove them if not used any more

File: tools/verify_partitions.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
+import argparse
```
Comment (Rhett-Ying) [inline tools/verify_partitions.py:1]: Now you're adding more and more scripts into `//tools`, we probably need to re-organize the layout for the scripts under `//tools` to have a clear hierarchy.

**Review by Rhett-Ying — APPROVED** [review]
generally looks good to me. pls take a look at the comments.

File: tools/verification_utils.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    _etype_str_to_tuple,
+    _etype_tuple_to_str,
+    _get_inner_edge_mask,
+    _get_inner_node_mask,
+    RESERVED_FIELD_DTYPE,
+)
+from distpartitioning.utils import get_idranges
+
+
+def read_pq_file(ff):
```
[the remark this answers is not in the document]
Comment (kylasa) [reply tools/verification_utils.py:24]: Done.

File: tools/verification_utils.py [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    """
+    return np.load(ff)
+
+
+def read_csv_file(ff, single_col=False):
+    """Read a csv file
+
+    Parameters:
+    -----------
+    ff : string
```
[the remark this answers is not in the document]
Comment (kylasa) [reply tools/verification_utils.py:63]: replaced with 'fname'

File: tools/verify_partitions.py (hunk 3 above)
Comment (kylasa) [reply tools/verify_partitions.py:1]: As discussed in our late night call on Wednesday, we agreed that re-organizing the whole tests/tools directory is a time consuming task and should be done with due diligence. We agreed that this task can be taken-after the pipeline is made available for general public.

File: tests/tools/test_dist_part.py (hunk 1 above)
Comment (kylasa) [reply tests/tools/test_dist_part.py]: This code should be deprecated using a separate PR#5310.

File: tests/tools/test_dist_part.py (hunk 2 above)
Comment (kylasa) [reply tests/tools/test_dist_part.py]: This code will changed appropriately using PR #5310

**Pull request merged**